src/ext/config.js

```diff
@@ -179,8 +180,16 @@ function hcBaseConfig() {
         "-----END PUBLIC KEY-----";
     hcConfig["spending-restrictions"]["status-code"] = [200];
     hcConfig["spend-action"]["redeem-method"] = "no-reload";
-    hcConfig["spend-action"]["urls"] = ["https://*.hcaptcha.com/getcaptcha", "https://*.hmt.ai/getcaptcha", "http://127.0.0.1/getcaptcha"];
-    hcConfig["issue-action"]["urls"] = ["https://*.hcaptcha.com/checkcaptcha/*", "https://*.hmt.ai/checkcaptcha/*", "http://127.0.0.1/checkcaptcha/*"];
+    hcConfig["spend-action"]["urls"] = [
+        /^https:\/\/(.*\.)?hcaptcha\.com\/getcaptcha(\?.*)?$/,
+        /^https:\/\/(.*\.)?hmt\.ai\/getcaptcha(\?.*)?$/,
+        /^https:\/\/127\.0\.0\.1\/getcaptcha(\?.*)?$/,
+    ];
+    hcConfig["issue-action"]["urls"] = [
+        /^https:\/\/(.*\.)?hcaptcha\.com\/checkcaptcha(\/.*)?$/,
+        /^https:\/\/(.*\.)?hmt\.ai\/checkcaptcha(\/.*)?$/,
```

Comment:
`.*` might also match `https://example.com/.hmt.ai/checkcaptcha`, let's try to avoid that with this tweak:
```suggestion
        /^https:\/\/([^/]*\.)?hmt\.ai\/checkcaptcha(\/.*)?$/,
```

This applies to all previous regexes as well. Previously wildcards were replaced by `[^\\/]*` via utils.js.